src/modules/fancyzones/FancyZonesLib/WindowUtils.cpp

```diff
@@ -0,0 +1,448 @@
+    {
+        return true;
+    }
+    
+    static std::vector<std::wstring> defaultExcludedApps = { NonLocalizable::PowerToysAppFZEditor, NonLocalizable::CoreWindow, NonLocalizable::SearchUI };
+    return (find_app_name_in_path(processPath, defaultExcludedApps));
+}
+
+void FancyZonesWindowUtils::SwitchToWindow(HWND window) noexcept
+{
+    // Check if the window is minimized
+    if (IsIconic(window))
+    {
+        // Show the window since SetForegroundWindow fails on minimized windows
+        ShowWindow(window, SW_RESTORE);
+    }
+
+    // This is a hack to bypass the restriction on setting the foreground window
+    INPUT inputs[1] = { { .type = INPUT_MOUSE } };
+    SendInput(ARRAYSIZE(inputs), inputs, sizeof(INPUT));
+
+    SetForegroundWindow(window);
+}
+
+void FancyZonesWindowUtils::SizeWindowToRect(HWND window, RECT rect) noexcept
+{
+    WINDOWPLACEMENT placement{};
+    ::GetWindowPlacement(window, &placement);
+
+    // Wait if SW_SHOWMINIMIZED would be removed from window (Issue #1685)
+    for (int i = 0; i < 5 && (placement.showCmd == SW_SHOWMINIMIZED); ++i)
+    {
+        std::this_thread::sleep_for(std::chrono::milliseconds(100));
+        ::GetWindowPlacement(window, &placement);
+    }
+
+    // Do not restore minimized windows. We change their placement though so they restore to the correct zone.
+    if ((placement.showCmd != SW_SHOWMINIMIZED) &&
+        (placement.showCmd != SW_MINIMIZE))
+    {
+        placement.showCmd = SW_RESTORE;
+    }
+
+    // Remove maximized show command to make sure window is moved to the correct zone.
+    if (placement.showCmd == SW_SHOWMAXIMIZED)
+    {
+        placement.showCmd = SW_RESTORE;
+        placement.flags &= ~WPF_RESTORETOMAXIMIZED;
+    }
+
+    ScreenToWorkAreaCoords(window, rect);
+
+    placement.rcNormalPosition = rect;
+    placement.flags |= WPF_ASYNCWINDOWPLACEMENT;
+
+    ::SetWindowPlacement(window, &placement);
+    // Do it again, allowing Windows to resize the window and set correct scaling
+    // This fixes Issue #365
+    ::SetWindowPlacement(window, &placement);
+}
+
+void FancyZonesWindowUtils::SaveWindowSizeAndOrigin(HWND window) noexcept
+{
+    HANDLE handle = GetPropW(window, ZonedWindowProperties::PropertyRestoreSizeID);
+    if (handle)
+    {
+        // Size already set, skip
+        return;
+    }
+
+    RECT rect;
+    if (GetWindowRect(window, &rect))
+    {
+        int width = rect.right - rect.left;
+        int height = rect.bottom - rect.top;
+        int originX = rect.left;
+        int originY = rect.top;
+
+        DPIAware::InverseConvert(MonitorFromWindow(window, MONITOR_DEFAULTTONULL), width, height);
+        DPIAware::InverseConvert(MonitorFromWindow(window, MONITOR_DEFAULTTONULL), originX, originY);
+
+        std::array<int, 2> windowSizeData = { width, height };
+        std::array<int, 2> windowOriginData = { originX, originY };
+        HANDLE rawData;
+        memcpy(&rawData, windowSizeData.data(), sizeof rawData);
```

Comment:
Question: Does `rawData` have a size at this specific point? If not, then `memcopy(x, x, sizeof rawData)` might be of size null? Shouldn't it be `sizeof windowSizeData.data()`?

Reply:
`sizeof(HANDLE)` is 64 on the 64-bit system, it's OK there.